docs/cookbook/custom-email.rst

```diff
@@ -0,0 +1,176 @@
+    {% endblock %}
+
+    {% block body %}
+        {% autoescape %}
+            The {{ variant.name }} variant is out of stock!
+        {% endautoescape %}
+    {% endblock %}
+
+* configure the email under ``sylius_mailer:`` in the ``app/config/emails.yml`` included in ``app/config/yml``.
+
+.. code-block:: yaml
+
+    # app/config/emails.yml
+    sylius_mailer:
+        sender:
+            name: Example.com
+            address: no-reply@example.com
+        templates:
+            app.email.order_confirmation.name: "AppBundle:Email:out_of_stock.html.twig"
+        emails:
+            out_of_stock:
+                subject: "A product has become out of stock!"
+                template: "AppBundle:Email:out_of_stock.html.twig"
+
+.. code-block:: yaml
+
+    # app/config/config.yml
+    imports:
+        - { resource: "emails.yml" }
+
+2. Create an Email Manager class:
+---------------------------------
+
+* It will need the **EmailSender**, the **AvailabilityChecker** and the **AdminUser Repository**.
+* It will operate on the **Order** where it needs to check each OrderItem, get their **ProductVariants** from Units and check if they are available.
+
+.. code-block:: php
+
+    <?php
+
+    namespace AppBundle\EmailManager;
+
+    use Sylius\Component\Core\Model\OrderInterface;
+    use Sylius\Component\Inventory\Checker\AvailabilityCheckerInterface;
+    use Sylius\Component\Mailer\Sender\SenderInterface;
+    use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+    class OutOfStockEmailManager
+    {
+        /**
+         * @var SenderInterface
+         */
+        private $emailSender;
+
+        /**
+         * @var AvailabilityCheckerInterface $availabilityChecker
+         */
+        private $availabilityChecker;
+
+        /**
+         * @var RepositoryInterface $adminUserRepository
+         */
+        private $adminUserRepository;
+
+        /**
+         * @param SenderInterface $emailSender
+         * @param AvailabilityCheckerInterface $availabilityChecker
+         * @param RepositoryInterface $adminUserRepository
+         */
+        public function __construct(
+            SenderInterface $emailSender,
+            AvailabilityCheckerInterface $availabilityChecker,
+            RepositoryInterface $adminUserRepository
+        ) {
+            $this->emailSender = $emailSender;
+            $this->availabilityChecker = $availabilityChecker;
+            $this->adminUserRepository = $adminUserRepository;
+        }
+
+        /**
+         * @param OrderInterface $order
+         */
+        public function sendOutOfStockEmail(OrderInterface $order)
+        {
+            $admins = $this->adminUserRepository->findAll();
+
+            foreach($order->getItems() as $item) {
+                $unit = $item->getUnits()->first();
+                $variant = $unit->getStockable();
+
+                $outOfStock = $this->availabilityChecker->isStockSufficient($variant, 1);
+
+                if ($outOfStock) {
```

Comment:
$outOfStock is true when we have sufficient product's elements quantity in inventory, right? Shouldn't this variable has name like `inStock` or something like that? Because now I think, that when it is "out of Stock" then it should continue istead of send an email with notification.